src/libultra/os/getmemsize.c

```diff
@@ -3,13 +3,13 @@
 #define STEP 0x100000
 
 u32 osGetMemSize(void) {
-    u32* ptr;
+    volatile u32* ptr;
     u32 size = 0x400000;
     u32 data0;
     u32 data1;
 
     while (size < 0x800000) {
-        ptr = (u32*)(0xA0000000 + size);
+        ptr = (volatile u32*)(0xA0000000 + size);
```

Comment:
```suggestion
        ptr = OS_PHYSICAL_TO_K1(size);
```
Alternatives include
`ptr = (volatile u32*)(K1BASE + size);`
`ptr = (void*)(K1BASE + size);`

Reply:
tough choice
I think `K1BASE + ` since it's constructed an address rather than converting from physical but it's kind of whatever
And why not repeat `vu32*`

Reply:
`vu32*` as the cast is also an option

Reply:
Yeah I meant "why not" as in "I'll do that, why not"